src/python/pants/init/engine_initializer.py

```diff
@@ -378,15 +375,15 @@ def union_membership_singleton() -> UnionMembership:
 
     @rule
     def build_root_singleton() -> BuildRoot:
-      return BuildRoot.instance
+      return cast(BuildRoot, BuildRoot.instance)
```

Comment:
Does `BuildRoot.instance` not return a `BuildRoot` yet?

Hm, after looking in `meta.py`, I see: https://github.com/pantsbuild/pants/blob/fd469cb62f330b73e7814f86e94558d16cfc9da3/src/python/pants/util/meta.py#L25-L27

It might be good to add a `TODO` here then as well? Or, if you agree, I can make that into an issue and we can link to it here?

Reply:
Correct, `.instance` does not work yet. I'll create an issue right now but going to avoid adding a `TODO` because MyPy will tell you when the `cast` is redundant as proof that `meta.py` is fixed :) Want to avoid a possibly stale comment.